src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,286 @@
+    } else {
         func_8029ED98(0, 1.0f);
+    }
+    
+    /*
+     * Sets fish groups forward velocity to a random float.
+     * Assigns random floats to fish groups UnkFC and Unk104, value dependant on level.
+     */
     if (o->oTimer == 0) {
         o->oForwardVel = RandomFloat() * 2 + 3.0f;
-        if (gCurrLevelNum == LEVEL_SA)
+        if (gCurrLevelNum == LEVEL_SA) {
             o->oFishGroupUnkFC = RandomFloat() * 700.0f;
-        else
+        } else {
             o->oFishGroupUnkFC = RandomFloat() * 100.0f;
+        }
         o->oFishGroupUnk104 = RandomFloat() * 500 + 200.0f;
     }
+    
+    // Rotate fish towards mario at a rate of 5.625 degrees (0x400)
     o->oFishGroupUnkF8 = gMarioObject->oPosY + o->oFishGroupUnkFC;
     obj_rotate_yaw_toward(o->oAngleToMario, 0x400);
+    
+    // If fish groups are too close regroup
     if (o->oPosY < o->oFishGroupUnkF4 - 50.0f) {
-        if (sp1C < 0.0f)
-            sp1C = 0.0f - sp1C;
-        if (sp1C < 500.0f)
-            func_802BF59C(2);
-        else
-            func_802BF59C(4);
+        if (fishY < 0.0f) {
+            fishY = 0.0f - fishY;
+        }
+        if (fishY < 500.0f) {
+            fish_regroup(2);
+        } else {
+            fish_regroup(4);
+        }
     } else {
         o->oPosY = o->oFishGroupUnkF4 - 50.0f;
-        if (sp1C > 300.0f)
+        if (fishY > 300.0f) {
             o->oPosY = o->oPosY - 1.0f;
+        }
     }
-    if (o->oDistanceToMario < o->oFishGroupUnk104 + 150.0f)
+    
+    // Remove fish if its distance to mario is smaller than his distance to Unk104 + 150.0f
+    if (o->oDistanceToMario < o->oFishGroupUnk104 + 150.0f) {
         o->oAction = 2;
+    }
 }
 
-void ActionFishGroup2(void) {
-    f32 sp1C = o->oPosY - gMarioObject->oPosY;
-    s32 sp18;
+/*
+ * Moves fish in relation to distance from each other and Mario.
+ */
+void fish_group_act_move(void) {
+    f32 fishY = o->oPosY - gMarioObject->oPosY;
+    // Marked unused, but has arithmetic performed on it in a useless manner.
+    UNUSED s32 distance;
     o->oFishGroupUnkF8 = gMarioObject->oPosY + o->oFishGroupUnkFC;
+    // Set fish groups to random floats when timer reaches zero and plays sound effect.
     if (o->oTimer == 0) {
         o->oFishGroupUnk110 = RandomFloat() * 300.0f;
         o->oFishGroupUnk100 = RandomFloat() * 1024.0f + 1024.0f;
         o->oFishGroupUnk108 = RandomFloat() * 4.0f + 8.0f + 5.0f;
-        if (o->oDistanceToMario < 600.0f)
-            sp18 = 1;
-        else
-            sp18 = (s32)(1.0 / (o->oDistanceToMario / 600.0));
-        sp18 *= 127;
+        // variable distance, is unused.
+        if (o->oDistanceToMario < 600.0f) {
+            distance = 1;
+        } else {
+            distance = (s32)(1.0 / (o->oDistanceToMario / 600.0));
+        }
+        distance *= 127;
         PlaySound2(SOUND_GENERAL_MOVING_WATER);
     }
-    if (o->oTimer < LEVEL_SA)
+    // Enable fish animation. Parameters differ if oTimer < SA.
```

Comment:
This is technically a repo typo. LEVEL_SA should just be 20 here, it is fixed privately.